### PR TITLE
Testfix/more rest tests

### PR DESCRIPTION
**Pull request by PenghaiZhang**
The first thing these rest tests do is to login. However, in the new UI, these tests do not wait until the home page is completed loaded. As a result, they fail to open the oauthadmin page, and then fail to get oauth token.

### CODE REVIEW

File: autotest/OldTests/src/test/java/com/tle/webtests/test/webservices/rest/AbstractRestApiTest.java [hunk 1]
```diff
@@ -159,7 +158,6 @@ protected void cleanupAfterClass() throws Exception {
   }
 
   protected String getToken() throws IOException {
-    new WebDriverWait(context.getDriver(), 30).until((webDriver -> clients.size() >= 1));
```
Comment (edalex-ian) [inline autotest/OldTests/src/test/java/com/tle/webtests/test/webservices/rest/AbstractRestApiTest.java]: Interesting to change this, as it's the bit that @SammyIsConfused just recently added. But I guess I see why, as I was wondering before who `clients` was meant to increase in size, and you're highlighting the root cause here was the delay with the login page.

File: autotest/OldTests/src/test/java/com/tle/webtests/test/AbstractSessionTest.java [hunk 2]
```diff
@@ -28,7 +30,12 @@ public HomePage logon(String username, String password, PageContext context) {
   }
 
   public void logonOnly(String username, String password) {
-    new LoginPage(context).load().loginWithRedirect(username, password);
+    LoginPage loginPage = new LoginPage(context).load();
+    loginPage.loginWithRedirect(username, password);
+    // If using new UI, ensure the home page is completed loaded.
+    if (loginPage.usingNewUI()) {
```
Comment (edalex-ian) [inline autotest/OldTests/src/test/java/com/tle/webtests/test/AbstractSessionTest.java]: I wonder if we really need this `if` check. Potentially the comment with the line is sufficient, and in theory the line below will just pass straight away when in the Legacy UI mode.

Thoughts?

File: autotest/OldTests/src/test/java/com/tle/webtests/test/webservices/rest/AbstractRestApiTest.java (hunk 1 above)
Comment (PenghaiZhang) [reply autotest/OldTests/src/test/java/com/tle/webtests/test/webservices/rest/AbstractRestApiTest.java]: > Interesting to change this, as it's the bit that @SammyIsConfused just recently added. But I guess I see why, as I was wondering before who `clients` was meant to increase in size, and you're highlighting the root cause here was the delay with the login page.

If these tests fail to open oauthadmin page due to login delay, then `clients` will always be empty regardless of how long the tests wait here.

I got some key code here:

      for (OAuthClient clientInfo : oauthClients) {
        OAuthUtils.createClient(context, clientInfo);   ---this line fails due to oauthadmin not open
        clients.add(clientInfo);  --this line won't get executed
      }

File: autotest/OldTests/src/test/java/com/tle/webtests/test/AbstractSessionTest.java (hunk 2 above)
Comment (PenghaiZhang) [reply autotest/OldTests/src/test/java/com/tle/webtests/test/AbstractSessionTest.java]: > I wonder if we really need this `if` check. Potentially the comment with the line is sufficient, and in theory the line below will just pass straight away when in the Legacy UI mode.
> 
> Thoughts?

Yes, this `if` is redundant. I was thinking `eqpageForm` only exists in new UI but now I confirm it also exists in the Legacy UI. So let me remove this if.

**Pull request merged**